### PR TITLE
Add support for the sourcemaps ignorelist

### LINKED ISSUE #480: Add support for x_google_ignoreList extension

### PR DESCRIPTION
**Pull request by bomsy**
Add support for the x_google_ignoreList to the SourceMapConsumer. 
Fixes #480

### CODE REVIEW

File: test/test-source-map-consumer.js [hunk 1]
```diff
@@ -88,6 +88,16 @@ exports["test that the `sources` field has the original sources"] =
     map.destroy();
   };
 
+exports["test that the SourceMapConsumer supports the ignoreList"] = async function (assert) {
+    const map = await new SourceMapConsumer(util.testMapWithIgnoreList);
+    const sources = map.sources;
+    const ignoreList = map.x_google_ignoreList;
+    assert.equal(ignoreList[0], 0);
+    assert.equal(sources[ignoreList[0]], "/the/root/one.js");
+    assert.equal(ignoreList.length, 1);
```
Comment (ochameau) [inline test/test-source-map-consumer.js:97]: This assertion should be first.

**Comment by ochameau** [comment]
It would be nice to mention any reference regarding this new field.
May be chromium original implementation changeset? Any public document describing this? Any discussion about this field on github...?
I found [this doc](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?pli=1#heading=h.ghqpj1ytqjbm) for example.

Otherwise you are having linting issues reported by CI:
https://github.com/mozilla/source-map/actions/runs/4580007974/jobs/8088310426?pr=481

File: test/test-source-map-consumer.js (hunk 1 above)
Comment (bomsy) [reply test/test-source-map-consumer.js:97]: Sure.

**Comment by bomsy** [comment]
> It would be nice to mention any reference regarding this new field. May be chromium original implementation changeset? Any public document describing this? Any discussion about this field on github...? I found [this doc](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?pli=1#heading=h.ghqpj1ytqjbm) for example.

Oh yeah ...that's the spec link. So i updated the README documentation with the info. Since its in the sepc doument
>

**Comment by hildjj** [comment]
You added the `.DS_Store` file.  That should probably be in .gitignore.

**Pull request merged**